Core/include/Acts/TrackFitting/GainMatrixSmoother.hpp

```diff
@@ -53,98 +45,92 @@ class GainMatrixSmoother {
-        ACTS_VERBOSE("Jacobian:\n" << ts.jacobian());
-        ACTS_VERBOSE("Prev. predicted covariance\n"
-                     << prev_ts.predictedCovariance() << "\n, inverse: \n"
-                     << prev_ts.predictedCovariance().inverse());
-
-        // Gain smoothing matrix
-        // NB: The jacobian stored in a state is the jacobian from previous
-        // state to this state in forward propagation
-        G = ts.filteredCovariance() * prev_ts.jacobian().transpose() *
-            prev_ts.predictedCovariance().inverse();
-
-        if (G.hasNaN()) {
-          error = KalmanFitterError::SmoothFailed;  // set to error
-          return false;                             // abort execution
-        }
-
-        ACTS_VERBOSE("Gain smoothing matrix G:\n" << G);
-
-        ACTS_VERBOSE("Calculate smoothed parameters:");
-        ACTS_VERBOSE("Filtered parameters: " << ts.filtered().transpose());
-        ACTS_VERBOSE(
-            "Prev. smoothed parameters: " << prev_ts.smoothed().transpose());
-        ACTS_VERBOSE(
-            "Prev. predicted parameters: " << prev_ts.predicted().transpose());
-
-        // Calculate the smoothed parameters
-        ts.smoothed() =
-            ts.filtered() + G * (prev_ts.smoothed() - prev_ts.predicted());
-
-        ACTS_VERBOSE("Smoothed parameters are: " << ts.smoothed().transpose());
-
-        ACTS_VERBOSE("Calculate smoothed covariance:");
-        ACTS_VERBOSE("Prev. smoothed covariance:\n"
-                     << prev_ts.smoothedCovariance());
-
-        // And the smoothed covariance
-        ts.smoothedCovariance() =
-            ts.filteredCovariance() -
-            G * (prev_ts.predictedCovariance() - prev_ts.smoothedCovariance()) *
-                G.transpose();
-
-        // Check if the covariance matrix is semi-positive definite.
-        // If not, make one (could do more) attempt to replace it with the
-        // nearest semi-positive def matrix,
-        // but it could still be non semi-positive
-        BoundSymMatrix smoothedCov = ts.smoothedCovariance();
-        if (not detail::covariance_helper<BoundSymMatrix>::validate(
-                smoothedCov)) {
-          ACTS_DEBUG(
-              "Smoothed covariance is not positive definite. Could result in "
-              "negative covariance!");
-        }
-        // Reset smoothed covariance
-        ts.smoothedCovariance() = smoothedCov;
-        ACTS_VERBOSE("Smoothed covariance is: \n" << ts.smoothedCovariance());
-
-        prev_ts = ts;
-        return true;  // continue execution
-      });
-    }
-    if (error) {
-      // error is set, return result
-      return *error;
+      return Result<void>::success();
     }
 
-    // construct parameters from last track state
-    return Result<void>::success();
+    ACTS_VERBOSE("Start smoothing from previous track state at index: "
+                 << prev_ts.previous());
+
+    // default error represents an invalid non-error state, no need for optional
```

Comment:
This comment makes no sense out of context of the changeset.

Reply:
Fixed.